tests/PHPStan/Analyser/data/set-type-type-specifying.php

```diff
@@ -0,0 +1,511 @@
+<?php
+
+namespace SetTypeTypeSpecifying;
+
+use stdClass;
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param 'string'|'int'|'integer'|'float'|'double'|'bool'|'boolean'|'array'|'object'|'null' $castTo
+ */
+function setTypeSpecifying(string $value, string $castTo): void
```

Comment:
I think a 2nd test function which works with a non-constant $x would be helpful. 

Like

`function doFoo(int $i, string $s ...)`

And assert similar to what you did in this method

Reply:
If I caught the intention of your message correctly, this is just a mistake on my part when writing this test initially. The function was supposed to receive a mixed type for the first argument to match the signature of `settype`. I've changed this now, but if that wasn't the intention, please correct me.

Reply:
I would add a test-functions like

```
function doBoolean(int $i, string $s) {
	settype($i, 'boolean');
	assertType('bool', $x);

	settype($s, 'boolean');
	assertType('bool', $x);
}
```

since for this non-constant values, different types are returned.
there are lots of combinations and I think it doesn't make sense to test all combinations.. but at least a few most common ones should be tested

Reply:
Updated - I've added the functions as you described to the existing test cases.